pkg/store/gcs.go

```diff
@@ -572,11 +610,11 @@ func (r *gcsChunkReader) Chunk(id uint64) (chunks.Chunk, error) {
 
 	// We don't know the length of the chunk, load 1024 bytes, which should be enough
 	// even for long ones.
-	rd, err := obj.NewRangeReader(context.TODO(), int64(off), int64(off+1024))
+	rd, err := obj.NewRangeReader(r.ctx, int64(off), int64(off+1024))
 	if err != nil {
 		return nil, errors.Wrap(err, "create reader")
 	}
-	defer r.Close()
+	defer rd.Close()
```

Comment:
wow... This is the result of using short variable names ):  I did not notice that too during review. Good you found it!

Reply:
Yea, bad choice to chose two which are so similar. Will try to avoid/replace it in the future.